### PR TITLE
[GIT PULL] Integrating with Jekyll

### PR DESCRIPTION
**Pull request by kiraio-moe**
Hello :)

Done with integrating this repo with Jekyll. Hopefully it can be useful.
I'm not touching any of the design cause seems like [Amir Zuhdi Wibowo](https://t.me/cakadi190 "Amir Zuhdi Wibowo at Telegram") (he's here?) made a great design.

I want too but, let's watch others do XD
```
----------------------------------------------------------------
The following changes since commit c3e22f7832be820f1a81fd89e7b39083d46a0809:

  Merge branch 'sprtcrnbry-rev' (2021-05-22 21:14:53 +0700)

are available in the Git repository at:

  https://github.com/kiraio-moe/www.gnuweeb.org kiraio

for you to fetch changes up to 0db4bbd49f87cceb77b933423f782bacdc64128c:

  Fix Accordion section (2021-09-23 10:55:22 +0700)

----------------------------------------------------------------
Bayu Satiyo (6):
      Restructure Directory
      Jekyll Integration
      Setup Jekyll configuration
      Completing basic configuration
      Fully integrated with Jekyll
      Fix Accordion section

 .editorconfig                                          |  15 ++++++
 .gitignore                                             | 102 +++++++++++++++++++++++++++++++++++++
 CODE_OF_CONDUCT.md                                     |  51 +++++++++++++++++++
 Gemfile                                                |  14 +++++
 Gemfile.lock                                           |  87 +++++++++++++++++++++++++++++++
 README.md                                              |  40 +++++++++++++++
 _config.yml                                            |  50 ++++++++++++++++++
 _includes/footer.html                                  |   7 +++
 _includes/google-analytics.html                        |  17 +++++++
 _includes/head.html                                    |  18 +++++++
 _includes/header.html                                  |  22 ++++++++
 _includes/scripts.html                                 |   5 ++
 _includes/sns.html                                     |  13 +++++
 _includes/styles.html                                  |  10 ++++
 _layouts/default.html                                  |  17 +++++++
 _layouts/home.html                                     | 292 ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 assets/css/bootstrap/bootstrap.min.css                 |   7 +++
 assets/css/bootstrap/bootstrap.min.css.map             |   1 +
 {public/assets/fw => assets/css}/default/gnuweeb.css   |  34 ++++++++-----
 {public => assets/img}/favicon.png                     | Bin
 assets/js/bootstrap/bootstrap.bundle.min.js            |   7 +++
 assets/js/bootstrap/bootstrap.bundle.min.js.map        |   1 +
 bin/build                                              |  51 +++++++++++++++++++
 bin/check                                              |   9 ++++
 bin/clean                                              |  26 ++++++++++
 bin/serve                                              |   9 ++++
 bin/setup                                              |  13 +++++
 bin/update                                             |  13 +++++
 gnuweeb.gemspec                                        |  25 +++++++++
 index.md                                               |   8 +++
 logs/.gitignore                                        |   1 -
 public/assets/fw/bootstrap/bootstrap.min.css           |   7 ---
 public/assets/fw/bootstrap/bootstrap.min.css.map       |   1 -
 public/assets/js/bootstrap/bootstrap.bundle.min.js     |   7 ---
 public/assets/js/bootstrap/bootstrap.bundle.min.js.map |   1 -
 public/index.html                                      | 360 ---------------------------------------------------------------------------------------------------------------------------------
 36 files changed, 952 insertions(+), 389 deletions(-)
 create mode 100644 .editorconfig
 create mode 100644 .gitignore
 create mode 100644 CODE_OF_CONDUCT.md
 create mode 100644 Gemfile
 create mode 100644 Gemfile.lock
 create mode 100644 README.md
 create mode 100644 _config.yml
 create mode 100644 _includes/footer.html
 create mode 100644 _includes/google-analytics.html
 create mode 100644 _includes/head.html
 create mode 100644 _includes/header.html
 create mode 100644 _includes/scripts.html
 create mode 100644 _includes/sns.html
 create mode 100644 _includes/styles.html
 create mode 100644 _layouts/default.html
 create mode 100644 _layouts/home.html
 create mode 100644 assets/css/bootstrap/bootstrap.min.css
 create mode 100644 assets/css/bootstrap/bootstrap.min.css.map
 rename {public/assets/fw => assets/css}/default/gnuweeb.css (79%)
 rename {public => assets/img}/favicon.png (100%)
 create mode 100644 assets/js/bootstrap/bootstrap.bundle.min.js
 create mode 100644 assets/js/bootstrap/bootstrap.bundle.min.js.map
 create mode 100644 bin/build
 create mode 100644 bin/check
 create mode 100644 bin/clean
 create mode 100644 bin/serve
 create mode 100644 bin/setup
 create mode 100644 bin/update
 create mode 100644 gnuweeb.gemspec
 create mode 100644 index.md
 delete mode 100644 logs/.gitignore
 delete mode 100644 public/assets/fw/bootstrap/bootstrap.min.css
 delete mode 100644 public/assets/fw/bootstrap/bootstrap.min.css.map
 delete mode 100644 public/assets/js/bootstrap/bootstrap.bundle.min.js
 delete mode 100644 public/assets/js/bootstrap/bootstrap.bundle.min.js.map
 delete mode 100644 public/index.html
```

### CODE REVIEW

**Review by irvanmalik48 — DISMISSED** [review]
Nice. Approved.

**Comment by ammarfaizi2** [comment]
Please rebase your commits, make sure sign them with GPG.

**Comment by kiraio-moe** [comment]
On it.

**Comment by kiraio-moe** [comment]
Done, Sir!

**Comment by ammarfaizi2** [comment]
@kiraio-moe you changed your key?

The key you submitted is
```
D45791DEA507807645DD3CBDEC5875721E7D8E30 "Bayu Satiyo (kiraio) <itsyuukunz@gmail.com>"
```
(https://github.com/GNUWeeb/pgpkeys/commit/2303b802d25b3f09a02be8b314972fe0bdba528c)

But your commits have been signed with
```
AC058DD7FB403E1F5C48E5FBA09859B6599FFDF6
```

**Comment by kiraio-moe** [comment]
> @kiraio-moe you changed your key?
> 
> The key you submitted is
> 
> ```
> D45791DEA507807645DD3CBDEC5875721E7D8E30 "Bayu Satiyo (kiraio) <itsyuukunz@gmail.com>"
> ```
> 
> ([GNUWeeb/pgpkeys@2303b80](https://github.com/GNUWeeb/pgpkeys/commit/2303b802d25b3f09a02be8b314972fe0bdba528c))
> 
> But your commits have been signed with
> 
> ```
> AC058DD7FB403E1F5C48E5FBA09859B6599FFDF6
> ```

Just updated. Check it.

**Comment by ammarfaizi2** [comment]
Please drop these commits, I require every single commit to be signed.
```
2ef5a5f373448d237a0e45aecc58da4cde5e7d32 ("Fully integrated with Jekyll")
b7426982704fe7cf2a3e194077fbf18e5e8d668f ("Completing basic configuration")
db79da9aaf3becc46c82b2589be0068459484a6a ("Setup Jekyll configuration")
56c3da12b8f2edf341f75c003619d397e6eaabdb ("Jekyll Integration")
6804c424871dc08d670b08b2c4d7b2f39ad45561 ("Restructure Directory")
```

I did not ask you to add more signed commits, but I require you to rebase so that old commits are signed, not adding new commits. Once you have cleaned that up, we can re-review the contents.

**Review by ammarfaizi2 — CHANGES_REQUESTED** [review]
I am requesting for changes.

**Comment by kiraio-moe** [comment]
> Please drop these commits, I require every single commit to be signed.
> 
> ```
> 2ef5a5f373448d237a0e45aecc58da4cde5e7d32 ("Fully integrated with Jekyll")
> b7426982704fe7cf2a3e194077fbf18e5e8d668f ("Completing basic configuration")
> db79da9aaf3becc46c82b2589be0068459484a6a ("Setup Jekyll configuration")
> 56c3da12b8f2edf341f75c003619d397e6eaabdb ("Jekyll Integration")
> 6804c424871dc08d670b08b2c4d7b2f39ad45561 ("Restructure Directory")
> ```
> 
> I did not ask you to add more signed commits, but I require you to rebase so that old commits are signed, not adding new commits. Once you have cleaned that up, we can re-review the contents.

How to do that, sir?

Edit:
Just reading the documentation. In rebase mode change pick to drop?

**Comment by ammarfaizi2** [comment]
> How to do that, sir?
> 
> Edit:
> Just reading the documentation. In rebase mode change pick to drop?

Recall that the last commit before your commits is c3e22f7832be820f1a81fd89e7b39083d46a0809 ("Merge branch 'sprtcrnbry-rev'").

So your rebase command will be:
```
git rebase -i c3e22f7832be820f1a81fd89e7b39083d46a0809
```

It will spawn a text editor and show you this:
```
pick 772af78 Restructure Directory
pick 9ca7d07 Jekyll Integration
pick 3add683 Setup Jekyll configuration
pick 57c4aa9 Completing basic configuration
pick b388776 Fully integrated with Jekyll
pick 0db4bbd Fix Accordion section
pick 6804c42 Restructure Directory
pick 56c3da1 Jekyll Integration
pick db79da9 Setup Jekyll configuration
pick b742698 Completing basic configuration
pick 2ef5a5f Fully integrated with Jekyll
```

Change the `pick` to `drop` for the commits you want to drop.
Save it, you got it rebased.

**Comment by kiraio-moe** [comment]
> > How to do that, sir?
> > Edit:
> > Just reading the documentation. In rebase mode change pick to drop?
> 
> Recall that the last commit before your commits is [c3e22f7](https://github.com/GNUWeeb/www.gnuweeb.org/commit/c3e22f7832be820f1a81fd89e7b39083d46a0809) ("Merge branch 'sprtcrnbry-rev'").
> 
> So your rebase command will be:
> 
> ```
> git rebase -i c3e22f7832be820f1a81fd89e7b39083d46a0809
> ```
> 
> It will spawn a text editor and show you this:
> 
> ```
> pick 772af78 Restructure Directory
> pick 9ca7d07 Jekyll Integration
> pick 3add683 Setup Jekyll configuration
> pick 57c4aa9 Completing basic configuration
> pick b388776 Fully integrated with Jekyll
> pick 0db4bbd Fix Accordion section
> pick 6804c42 Restructure Directory
> pick 56c3da1 Jekyll Integration
> pick db79da9 Setup Jekyll configuration
> pick b742698 Completing basic configuration
> pick 2ef5a5f Fully integrated with Jekyll
> ```
> 
> Change the `pick` to `drop` for the commits you want to drop.
> Save it, you got it rebased.

After that, should I push or pull?

**Comment by ammarfaizi2** [comment]
> After that, should I push or pull?

You need to git push with force. Because you're re-writing history.

**Comment by ammarfaizi2** [comment]
Also recall that description should be word-wrapped 72 chars. I will let it pass this time, but for the future you need to do it properly.

What I mean is for commits:
```
b388776cdc1e9f9bd19bd5ef1161d94621884c97 ("Fully integrated with Jekyll")
57c4aa9f53524ef13dbf124cb6694fd64d4b13d3 ("Completing basic configuration")
3add683b146218cf50eec013e7f0e8c4d0161ce8 ("Setup Jekyll configuration")
```
The format for commit message is:
- First line is title
- Empty line
- Description (may be omited for trivial changes)
- Empty line again (if it has description)
- Then a `Signed-off-by` with your name and email.

Description should be word wrapped 72 chars. Some things should not be word-wrapped, they may be some kind of quoted text - long compiler error messages, oops reports, or whatever things that have a certain specific format.

**Comment by ammarfaizi2** [comment]
Let's wait for the approval from Irvan before merging this.

File: CODE_OF_CONDUCT.md
Comment (ammarfaizi2) [inline CODE_OF_CONDUCT.md:51]: We actually already have our own code of conduct, it's written here: https://t.me/GNUWeeb/460290

Comment (kiraio-moe) [reply CODE_OF_CONDUCT.md:51]: I have made a wiki page at my forked repo and contains what's on Telegram.
Will this wiki page be added to original repo?
Should I delete the CODE_OF_CONDUCT.md file?

**Comment by kiraio-moe** [comment]
> Description should be word wrapped 72 chars.

Is there any setting or extension to calculate it?

**Comment by ammarfaizi2** [comment]
> Is there any setting or extension to calculate it?

By default some text editors like vim, sublime text, etc will show you the column position where your cursor is currently pointing at.

You don't always have to use tool, but manually do it. Word-wrapping is a property of the text. And the tool you use to visualize things may not know. End result: you do word-wrapping at the only stage where you can do it, namely when writing it. Not when showing it.

**Comment by ammarfaizi2** [comment]
Extra tips from me for writing commit message:
1. Write your description on text editor which shows you the column position. So it's easy for you to word-wrap it.
2. When commit, use `git commit -am "Your commit message title" -se`. The `-e` flag will spawn a text editor for you to edit the commit message further.
3. Paste your description and save it.

**Review by ammarfaizi2 — APPROVED** [review]
I am done.
```
Reviewed-by: Ammar Faizi <ammarfaizi2@gmail.com>
```
Cc: @irvanmalik48

**Comment by kiraio-moe** [comment]
> Extra tips from me for writing commit message:
> 
>  1.  Write your description on text editor which shows you the column position. So it's easy for you to word-wrap it.
>  2.  When commit, use git commit -am "Your commit message title" -se. The -e flag will spawn a text editor for you to edit the commit message further.
>  3.  Paste your description and save it.

Noted. Thank you for the advice.

File: CODE_OF_CONDUCT.md
Comment (ammarfaizi2) [reply CODE_OF_CONDUCT.md:51]: > Will this wiki page be added to original repo?
> Should I delete the CODE_OF_CONDUCT.md file?

Ah, let it be there for the moment, I may clean that up later.

**Comment by ammarfaizi2** [comment]
Extra again from me, I train GNU/Weeb members to follow the Linux Kernel standard, including how to send pull request, how to commit, how to merge, etc.

Start from today, every pull request must contain diffstat and URL from where we can pull. The example can be seen here:
https://lore.kernel.org/lkml/172809ab-c9c4-fc36-6bba-3ea0128f748b@kernel.dk/

Plus the subject must have prefix [GIT PULL], so it's easier to filter from email inbox. I edited your description (added diffstat and git repo to pull).

**Comment by kiraio-moe** [comment]
How do you get this output?

>  .editorconfig                                          |  15 ++++++
>  .gitignore                                             | 102 +++++++++++++++++++++++++++++++++++++
>  CODE_OF_CONDUCT.md                                     |  51 +++++++++++++++++++
>  Gemfile                                                |  14 +++++
>  Gemfile.lock                                           |  87 +++++++++++++++++++++++++++++++
>  README.md                                              |  40 +++++++++++++++
>  _config.yml                                            |  50 ++++++++++++++++++
>  _includes/footer.html                                  |   7 +++
>  _includes/google-analytics.html                        |  17 +++++++
>  _includes/head.html                                    |  18 +++++++
>  _includes/header.html                                  |  22 ++++++++
>  _includes/scripts.html                                 |   5 ++
>  _includes/sns.html                                     |  13 +++++
>  _includes/styles.html                                  |  10 ++++
>  _layouts/default.html                                  |  17 +++++++
>  _layouts/home.html                                     | 292 ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
>  assets/css/bootstrap/bootstrap.min.css                 |   7 +++
>  assets/css/bootstrap/bootstrap.min.css.map             |   1 +
>  {public/assets/fw => assets/css}/default/gnuweeb.css   |  34 ++++++++-----

**Comment by ammarfaizi2** [comment]
> How do you get this output?

```
git checkout <your_branch_name>
git request-pull <start_commit> <your_repo_URL> <your_branch_name>
```

- `<start_commit>` is the hash of one commit before your first commit in the pull request.

For this pull request, the `<start_commit>` is
```
c3e22f7832be820f1a81fd89e7b39083d46a0809 ("Merge branch 'sprtcrnbry-rev'")
```

So your `request-pull` command will be:
```
git request-pull c3e22f7832be820f1a81fd89e7b39083d46a0809 https://github.com/kiraio-moe/www.gnuweeb.org kiraio
```
After that, you will get the pull request sentence and diffstat, copy them and put it on your GitHub pull request submission (with monospace styling, which means wrap it with triple backticks).

**Comment by ammarfaizi2** [comment]
Telegram ping @lappretard.

For Irvan (@irvanmalik48), we have a new way to merge pull request, don't use GitHub merge button to merge, start from today.

Please do the following:
```
# If you don't have the repo locally.
git clone https://github.com/GNUWeeb/www.gnuweeb.org;
cd www.gnuweeb.org;

git checkout master; # if you're not on branch master

# Pull this pull request (merge).
git pull --no-ff --log --sign https://github.com/kiraio-moe/www.gnuweeb.org kiraio;
```

The `--log` flag will give the description of what are being merged.

The git pull command will spawn a text editor, before your `Signed-off-by` tag, add this link and reviewed by tag
```
Link: https://github.com/GNUWeeb/www.gnuweeb.org/pull/17
Reviewed-by: Ammar Faizi <ammarfaizi2@gmail.com>
Reviewed-by: Irvan Malik <irvanmalik48@gmail.com>
Signed-off-by: Irvan Malik <irvanmalik48@gmail.com>
```

^ That's the extra part where Linus doesn't do in merge commit, but it's fine.

Save it, then:
```
git push origin master
```

The pull request on GitHub will automatically be merged as well.


From Linus:
```
"I notice that you have a GitHub merge commit in there, that's another
of those things that I *really* don't want to see - GitHub creates
absolutely useless garbage merges, and you should never ever use the
GitHub interfaces to merge anything [...]. GitHub is a perfectly fine
hosting site, and it does a number of other things well too, but
merges are not one of those things."

- Linus Torvalds
```

**Comment by irvanmalik48** [comment]
@ammarfaizi2 Alright, will follow suite accordingly

**Review by irvanmalik48 — APPROVED** [review]
Seems good, LGTM!

**Pull request merged**
